src/server/req_delete.c

```diff
@@ -416,7 +449,14 @@ req_deletejob(struct batch_request *preq)
 	else
 		qdel_mail = 1;
 
-	for (j = 0; j < count; j++) {
+	if (preq->rq_ind.rq_deletejoblist.rq_resume != 1) {
+		preply->brp_un.brp_deletejoblist.tot_jobs = count;
+		preply->brp_un.brp_deletejoblist.tot_arr_jobs = 0;
+		preply->brp_un.brp_deletejoblist.tot_rpys = 0;	
+	} else 
+		start_jobid = preq->rq_ind.rq_deletejoblist.jobid_to_resume;
```

Comment:
What happens if the start_jobid (in case of a resume) was deleted by another qdel request (or by job getting finished) by the time you resume the deletion?

Reply:
I have tried to simulate the behaviour in GDB, 
1) Put the breakpoint at the subsequent call of req_deletejob(). (resume request from work task)
2) In GDB,  removed the job object from which(start_jobid) the request is going to resume. (Similar to the situation job deleted by client or job getting finished)
3) While continuing gdb, the chk_job_request() function would return NULL since the job object is not available in the system and update the error code. 
4) Qdel client would get the response as UNKNOWN JOBID for that removed one.

Reply:
ok that sounds like it would work